macosx/Controller.m

```diff
@@ -3692,16 +3733,93 @@ - (NSRect) previewPanel: (QLPreviewPanel *) panel sourceFrameOnScreenForPreviewI
     }
 }
 
+- (IBAction) performTouchBarAction: (id) sender //senders here are always NSButtons, not NSTouchBarItems
+{
+    if (sender == createTouchBarButton)
+    {
+        [self createFile:sender];
+    }
+    if (sender == openFileTouchBarButton)
+    {
+        [self openShowSheet:sender];
+    }
+    if (sender == removeTouchBarButton)
+    {
+        [self removeNoDelete:sender];
+    }
+    if (sender == pauseAllTouchBarButton)
+    {
+        [self stopAllTorrents:sender];
+    }
+    if (sender == resumeAllTouchBarButton)
+    {
+        [self resumeAllTorrents:sender];
+    }
+    if (sender == speedLimitTouchBarButton)
+    {
+        [self toggleSpeedLimit: sender];
+    }
+    if (sender == cleanupTouchBarButton)
+    {
+        [self clearCompleted: sender];
+    }
+    if (sender == quickLookTouchBarButton)
+    {
+        [self toggleQuickLook:sender];
+    }
+    if (sender == showInfoTouchBarButton)
+    {
+        [self showInfo:sender];
+    }
+}
+
+- (void) syncToTouchBarForItem: (NSString *) ident withValue: (BOOL) result
```

Comment:
Are these buttons ever defined if the machine running doesn't have a touch bar? I think this method needs to be guarded by the `respondsToSelector` check above. `BOOL isTouchBarAvailable()`?